src/clients/inst_crowbar_patterns.rb

```diff
@@ -25,16 +25,51 @@
 #
 module Yast
   class InstCrowbarPatternsClient < Client
+    include Yast::Logger
     def main
 
       Yast.import "GetInstArgs"
+      Yast.import "Installation"
+      Yast.import "Kernel"
       Yast.import "PackagesProposal"
+      Yast.import "PackagesUI"
+      Yast.import "Pkg"
+      Yast.import "Stage"
 
       dialog_ret = :auto
       patterns = [ "cloud_admin" ]
 
-      unless GetInstArgs.going_back
-        Builtins.y2milestone("Selecting Cloud Admin pattern for installation...")
+      return dialog_ret if GetInstArgs.going_back
+
+      if Stage.normal
+        # In normal stage, we need to use the functionality from
+        # inst_add-on_software to setup source and target
+
+        Pkg.TargetInit(Installation.destdir, false)
+        Pkg.SourceStartManager(true)
+        patterns.each do |pattern|
+          Pkg.ResolvableInstall(pattern, :pattern)
+        end
+
+        ret = PackagesUI.RunPatternSelector
+        log.info("RunPatternSelector returned #{ret}")
+
+        dialog_ret = :next
+        dialog_ret = :abort if ret == :cancel
+
+        if ret == :accept || ret == :ok
+          # Add-on requires packages to be installed right now
+          log.info("Selected resolvables will be installed now")
```

Comment:
<sub>Really, really small nit picking :wink: : the convention is to not use brackets for the logging calls (like in the linked documentation), it's similar to `puts "foo"`.</sub>